images/airflow/2.9.2/python/mwaa/entrypoint.py

```diff
@@ -610,6 +615,16 @@ def run_airflow_command(cmd: str, environ: Dict[str, str]):
                 ]
             )
 
+        # Hybrid runs the scheduler and celery worker processes is a single container.
+        case "hybrid":
+            # The Sidecar healthcheck is currently limited to one healthcheck per port
+            # so the hybrid container can only include healthchecks for one subprocess.
+            # Only the worker healcheck conditions are enabled to monitor container health, so
+            # we pass an empty list of conditions to the scheduler process.
+            scheduler_subprocesses = _get_airflow_scheduler_subprocesses(environ, [])
+            worker_subprocesses = _get_airflow_worker_subprocesses(environ)
+            run_subprocesses(worker_subprocesses + scheduler_subprocesses)
```

Comment:
Since the worker subprocess has the health conditions attached, it needs to be made an essential process. Otherwise, if the worker exits due to failed healthcheck and scheduler is still running then the task will not be restarted but worker is lost.

Reply:
Regardless of health conditions, any process can die for any reason, so yes the scheduler and worker should both be considered essential processes, to ensure that if one fails, the other also exits so we get the container restarted.

Having said that, the implication of this is that if the scheduler fails for any reason, it will result in forcefully terminating the worker within [90 seconds](https://github.com/aws/amazon-mwaa-docker-images/blob/main/images/airflow/2.9.2/python/mwaa/subprocess/subprocess.py#L31-L33), which can result in task interruption. Thus, at the least we should:

1. Consider increasing the patience period for the worker to give tasks some time to finish. Obviously, we have to be careful how big this value we make it, as the downside is that it means during the patience period, we don't have a scheduler running.
2. Make sure our customers are aware of this.

The ideal solution, though, is to have the different processes restart independent of each other, i.e. doesn't cause a container restart. One idea that I wanted to work on but didn't have the time for is to implement a RetryableSubprocess class (which inherits from the Subprocess class). As the name suggests, this class restart the underlying subprocess if it fails. Then, when you run the workers and schedulers in a hybrid container, you use a RetryableSubprocess instead of Subprocess, meaning that if the scheduler/worker fails, it will retried "individually", up to a certain number of times after which we give up and accept that we will have to terminate running tasks.

Reply:
Good point. I will make both processes essential and increase the patience interval for the worker to 120s. Getting the base functionality implemented here will make end-to-end testing easier, and I can follow up with the implementation of the RetryableSubprocess to improve the experience here.